core/services/vrf/delegate.go

```diff
@@ -1,31 +1,376 @@
+	lsn.l.Infow("VRFListener: listening for run requests",
+		"maxUnconfirmed", lsn.cfg.EthMaxQueuedTransactions(),
+		"gasLimit", lsn.cfg.EthGasLimitDefault(),
+		"minConfs", minConfs)
+	for {
+		select {
+		case <-lsn.chStop:
+			unsubscribeLogs()
+			lsn.waitOnStop <- struct{}{}
+			return
+		case <-lsn.mbLogs.Notify():
+			// Process all the logs in the queue if one is added
+			for {
+				i, exists := lsn.mbLogs.Retrieve()
+				if !exists {
+					break
+				}
+				lb, ok := i.(log.Broadcast)
+				if !ok {
+					panic(fmt.Sprintf("VRFListener: invariant violated, expected log.Broadcast got %T", i))
+				}
+				alreadyConsumed, err := lsn.logBroadcaster.WasAlreadyConsumed(lsn.db, lb)
+				if err != nil {
+					lsn.l.Errorw("VRFListener: could not determine if log was already consumed", "error", err, "txHash", lb.RawLog().TxHash)
+					continue
+				} else if alreadyConsumed {
+					continue
+				}
+				req, err := lsn.coordinator.ParseRandomnessRequest(lb.RawLog())
+				if err != nil {
+					lsn.l.Errorw("VRFListener: failed to parse log", "err", err, "txHash", lb.RawLog().TxHash)
+					lsn.l.ErrorIf(lsn.logBroadcaster.MarkConsumed(lsn.db, lb), "failed to mark consumed")
+					continue
+				}
+
+				// Check if the vrf req has already been fulfilled
+				callback, err := lsn.coordinator.Callbacks(nil, req.RequestID)
+				if err != nil {
+					lsn.l.Errorw("VRFListener: unable to check if already fulfilled, processing anyways", "err", err, "txHash", req.Raw.TxHash)
+				} else if utils.IsEmpty(callback.SeedAndBlockNum[:]) {
+					// If seedAndBlockNumber is zero then the response has been fulfilled
+					// and we should skip it
+					lsn.l.Infow("VRFListener: request already fulfilled", "txHash", req.Raw.TxHash)
+					lsn.l.ErrorIf(lsn.logBroadcaster.MarkConsumed(lsn.db, lb), "failed to mark consumed")
+					continue
+				}
+
+				s := time.Now()
+				vrfCoordinatorPayload, req, err := lsn.ProcessLog(req, lb)
+				f := time.Now()
+				err = postgres.GormTransactionWithDefaultContext(lsn.db, func(tx *gorm.DB) error {
+					if err == nil {
+						// No errors processing the log, submit a transaction
+						var etx models.EthTx
+						var from common.Address
+						from, err = lsn.gethks.GetRoundRobinAddress()
+						if err != nil {
+							return err
+						}
+						etx, err = bulletprooftxmanager.CreateEthTransaction(tx,
+							from,
+							lsn.coordinator.Address(),
+							vrfCoordinatorPayload,
+							lsn.cfg.EthGasLimitDefault(),
+							lsn.cfg.EthMaxQueuedTransactions(),
+							&models.EthTxMetaV2{
+								JobID:         lsn.job.ID,
+								RequestID:     req.RequestID,
+								RequestTxHash: lb.RawLog().TxHash,
+							})
+						if err != nil {
+							return err
+						}
+						// TODO: Once we have eth tasks supported, we can use the pipeline directly
+						// and be able to save errored proof generations. Until then only save
+						// successful runs and log errors.
```

Comment:
Is there a story to track?

Reply:
Made one https://app.clubhouse.io/chainlinklabs/story/10934/use-ethtx-tasks-in-jpv2-vrf